김현수/mvvmstudy/app/src/main/java/com/example/mvvmstudy/data/MainRepository.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.mvvmstudy.data
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.mvvmstudy.Network.RetrofitClass
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MainRepository {
+
+    private val repos = mutableListOf<Repository>()
+
+    private val _repositoryList = MutableLiveData<List<Repository>>()
+    val repositoryList: LiveData<List<Repository>>
+        get() = _repositoryList
+
+    fun searchRepositories(query: String, page: String) {
+        RetrofitClass.retrofitService
+            .searchRepositories(query, page, 50)
+            .enqueue(object : Callback<Repositories> {
+                override fun onResponse(
+                    call: Call<Repositories>,
+                    response: Response<Repositories>
+                ) {
+                    when (response?.code() ?: 0) {
```

Comment:
response가 nullable이 아니여서 null check 할 필요가 없어졌네요!!
```suggestion
                    when (response.code()) {
```